lib/gateway.js

```diff
@@ -281,6 +281,22 @@ async function buildGateway (gatewayOpts, app) {
     schema,
     serviceMap,
     entityResolversFactory: factory,
+    pollingInterval: gatewayOpts.pollingInterval,
+    async refresh () {
+      for (const service of Object.values(serviceMap)) {
+        try {
+          await service.refresh()
+        } catch (err) {
+          continue
+        }
+      }
+
+      const serviceSDLs = Object.values(serviceMap).map(service => service.schemaDefinition)
+      const schema = buildFederatedSchema(serviceSDLs.join(''), true)
```

Comment:
Can we cache this output and avoid replacing the schema if the `serviceSDLs.join('')` did not change?

Reply:
I have added a simple cache for it

```js
if (this._serviceSDLs === undefined) {
  this._serviceSDLs = serviceSDLs.join('')
}
```

If we need to use LRU or something else, tell me and I will change it.